i18n-helpers/src/lib.rs

```diff
@@ -923,15 +1047,81 @@ The document[^1] text.
             > Postamble",
             vec![
                 (1, "Preamble"),
-                (
-                    2,
-                    "```rust\nfn hello() {\n    some_code()\n\n    // FIXME: do something here!\n    todo!()\n}\n```",
-                ),
+                (6, "// FIXME: do something here!\n"),
                 (10, "Postamble"),
             ],
         );
     }
 
+    #[test]
+    fn extract_messages_code_block_with_block_comment() {
+        assert_extract_messages(
+            "```rust\n\
+            /* block comment\n\
+             * /* nested block comment\n\
+             * */\n\
+             * \n\
+             * \n\
+             * \n\
+             * */\n\
+            ```\n",
+            vec![(
+                2,
+                "/* block comment\n* /* nested block comment\n* */\n* \n* \n* \n* */",
+            )],
+        );
+    }
+
+    #[test]
+    fn extract_messages_code_block_with_continuous_line_comments() {
+        assert_extract_messages(
+            r"```rust
+// continuous
+// line
+// comments
+{
+    // continuous
+    // line
+    // comments
+    let a = 1; // single line comment
+    let b = 1; // single line comment
+}
+```",
+            vec![
+                (2, "// continuous\n// line\n// comments\n"),
+                (6, "// continuous\n    // line\n    // comments\n"),
+                (9, "// single line comment\n"),
+                (10, "// single line comment\n"),
+            ],
+        );
+    }
+
+    #[test]
+    fn extract_messages_multi_language_code_blocks() {
+        assert_extract_messages(
+            "```c\n\
+            // C\n\
+            'C'; \"C\";\n\
```

Comment:
I think using a raw and un-indented string could be easier to read here.

I'm personally torn between using the `\n\` syntax and raw strings: I like the `\n\` syntax for short strings that don't have leading whitespace. For larger strings, I think we should use raw strings.